eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/TxPublisher.scala

```diff
@@ -86,7 +85,7 @@ object TxPublisher {
     def apply(txInfo: TransactionWithInputInfo, fee: Satoshi, parentTx_opt: Option[ByteVector32]): PublishFinalTx = PublishFinalTx(txInfo.tx, txInfo.input.outPoint, txInfo.desc, fee, parentTx_opt)
   }
   /** Publish an unsigned transaction that can be RBF-ed. */
-  case class PublishReplaceableTx(txInfo: ReplaceableTransactionWithInputInfo, commitments: Commitments) extends PublishTx {
+  case class PublishReplaceableTx(txInfo: ReplaceableTransactionWithInputInfo, commitments: Commitments, confirmationTarget: BlockHeight) extends PublishTx {
```

Comment:
That's an interesting point, it could make sense. In that case, would you define a new configuration section for those (instead of reusing the `eclair.on-chain-fees.target-blocks` like I'm doing)?

Reply:
> I believe it has to be non-optional

That's the thing, trying to decide on this leads to questioning the difference between `confirmationTarget` and `FeeTargets`. Making it mandatory means that the field will sometimes mean a "hard deadline", and sometime a "loose time preference". What is your rationale for making it mandatory?

Maybe the real difference is whether the transaction pays directly to our wallet (a.k.a. `Claim*Tx` transactions). With that heuristic:
- all `HtlcTx` would have a `Some(deadline)`
- all `Claim*Tx` would have `None` 

At this point I don't have enough context to bring more to the discussion.

> Does that sound good?

If we decide on making `ReplaceableTransactionWithInputInfo.confirmationTarget` mandatory, then yes your plan sounds good. One thing though: IIUC it will make us commit to a `confirmationTarget` once and for all when we generate a `Claim*Tx`, even if we later change our `eclair.on-chain-fees.target-blocks` before the transactions confirm. And that will be inconsistent with other non-replaceable transactions.

Reply:
What I have done (but not pushed yet), is to change `ReplaceableTransactionWithInputInfo` to:

```scala
sealed trait ReplaceableTransactionWithInputInfo extends TransactionWithInputInfo {
  /** Block before which the transaction must be confirmed. */
  def confirmBefore: BlockHeight
}
```

The rationale is that whenever you create such a transaction, you know when it should be confirmed (or at least, for commit txs without htlcs, you choose such a block height, because you have to choose something to aim for).

It's true that it's semantically less strong for commit txs without claimable htlcs, but since that same commit tx can later get a "hard deadline" (once there is an htlc we can claim), I don't think we can encode that difference in the types (not sure if that's easy to understand, let me know).

> all Claim*Tx would have None

Why? `ClaimHtlc*Tx` are exactly the same as `Htlc*Tx` here, they have a "hard" deadline.

> IIUC it will make us commit to a confirmationTarget once and for all when we generate a Claim*Tx, even if we later change our eclair.on-chain-fees.target-blocks before the transactions confirm. And that will be inconsistent with other non-replaceable transactions.

Not exactly, because we re-generate every tx in the `XxxCommitPublished` whenever the closing state is updated (and whenever we restart), so that would allow us to aim for something shorter than our previously broadcast transaction.

Reply:
> The rationale is that whenever you create such a transaction [...] you have to choose something to aim for.

Of course! A replaceable transaction must have always have a confirmation target, otherwise what's the point? Makes perfect sense.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2113/commits/3cd7b83c3b5de4777b922cf3af6483c5336f6c7f